test/http2_test/test_max_streams.py

```diff
@@ -0,0 +1,31 @@
+import logging
+import http2_base_server
+from hyperframe.frame import SettingsFrame
+
+class TestcaseSettingsMaxStreams(object):
+  """
+    This test sets MAX_CONCURRENT_STREAMS to 1 and asserts that at any point
+    only 1 stream is active.
```

Comment:
We should also specify somewhere that the client should first make a single RPC and block on the response before initiating this test. The C++ client can/will otherwise send 10 concurrent streams and complete before it receives the MAX_CONCURRENT_STREAMS setting from the server.